### PR TITLE
Nikos/4968/ens documentation

### LINKED ISSUE #4968: ENS Documentation
`web3-eth-ens` exports:

_All but `checkNetwork` are previously documented methods in `1.x`_

- [registryAddresses](https://github.com/web3/web3.js/blob/c2a845747032bfb22492f22634f0644b4774ecf9/packages/web3-eth-ens/src/config.ts#L39)
    - Simple object that maps network names to contract addresses
- `ENS` class
    - Public methods:
        - [getResolver](https://github.com/web3/web3.js/blob/c2a845747032bfb22492f22634f0644b4774ecf9/packages/web3-eth-ens/src/ens.ts#L64)
        - [setResolver](https://github.com/web3/web3.js/blob/c2a845747032bfb22492f22634f0644b4774ecf9/packages/web3-eth-ens/src/ens.ts#L71)
        - [setSubnodeRecord](https://github.com/web3/web3.js/blob/c2a845747032bfb22492f22634f0644b4774ecf9/packages/web3-eth-ens/src/ens.ts#L86)
        - [setApprovalForAll](https://github.com/web3/web3.js/blob/c2a845747032bfb22492f22634f0644b4774ecf9/packages/web3-eth-ens/src/ens.ts#L109)
        - [isApprovedForAll](https://github.com/web3/web3.js/blob/c2a845747032bfb22492f22634f0644b4774ecf9/packages/web3-eth-ens/src/ens.ts#L120)
        - [recordExists](https://github.com/web3/web3.js/blob/c2a845747032bfb22492f22634f0644b4774ecf9/packages/web3-eth-ens/src/ens.ts#L131)
        - [setSubnodeOwner](https://github.com/web3/web3.js/blob/c2a845747032bfb22492f22634f0644b4774ecf9/packages/web3-eth-ens/src/ens.ts#L138)
        - [getTTL](https://github.com/web3/web3.js/blob/c2a845747032bfb22492f22634f0644b4774ecf9/packages/web3-eth-ens/src/ens.ts#L151)
        - [setTTL](https://github.com/web3/web3.js/blob/c2a845747032bfb22492f22634f0644b4774ecf9/packages/web3-eth-ens/src/ens.ts#L158)
        - [getOwner](https://github.com/web3/web3.js/blob/c2a845747032bfb22492f22634f0644b4774ecf9/packages/web3-eth-ens/src/ens.ts#L169)
        - [setOwner](https://github.com/web3/web3.js/blob/c2a845747032bfb22492f22634f0644b4774ecf9/packages/web3-eth-ens/src/ens.ts#L176)
        - [setRecord](https://github.com/web3/web3.js/blob/c2a845747032bfb22492f22634f0644b4774ecf9/packages/web3-eth-ens/src/ens.ts#L188)
        - [setAddress](https://github.com/web3/web3.js/blob/c2a845747032bfb22492f22634f0644b4774ecf9/packages/web3-eth-ens/src/ens.ts#L201)
        - [setPubkey](https://github.com/web3/web3.js/blob/c2a845747032bfb22492f22634f0644b4774ecf9/packages/web3-eth-ens/src/ens.ts#L213)
        - [setContenthash](https://github.com/web3/web3.js/blob/c2a845747032bfb22492f22634f0644b4774ecf9/packages/web3-eth-ens/src/ens.ts#L225)
        - [getAddress](https://github.com/web3/web3.js/blob/c2a845747032bfb22492f22634f0644b4774ecf9/packages/web3-eth-ens/src/ens.ts#L238)
        - [getPubkey](https://github.com/web3/web3.js/blob/c2a845747032bfb22492f22634f0644b4774ecf9/packages/web3-eth-ens/src/ens.ts#L245)
        - [getContenthash](https://github.com/web3/web3.js/blob/c2a845747032bfb22492f22634f0644b4774ecf9/packages/web3-eth-ens/src/ens.ts#L252)
        - [checkNetwork](https://github.com/web3/web3.js/blob/c2a845747032bfb22492f22634f0644b4774ecf9/packages/web3-eth-ens/src/ens.ts#L260)
        - [supportsInterface](https://github.com/web3/web3.js/blob/c2a845747032bfb22492f22634f0644b4774ecf9/packages/web3-eth-ens/src/ens.ts#L293)

### PR DESCRIPTION
**Pull request by nikoulai**
## Description

closes #4968 

## Checklist for 4.x:

- [ ] I have selected the correct 4.x base branch.
- [ ] Within the description, the feature or issue is discussed in detail or an issue is linked.   
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added any required tests for the changes I made 
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `yarn` successfully
- [ ] I ran `yarn lint` successfully
- [ ] I ran `yarn build:web` successfully
- [ ] I ran `yarn test:unit` successfully
- [ ] I ran `yarn test:integration` successfully
- [ ] I ran `compile:contracts` successfully
- [ ] I have tested my code.
- [ ] I have updated the corresponding `CHANGELOG.md` file in the packages I have edited.

### CODE REVIEW

File: packages/web3-eth-ens/src/config.ts [hunk 1]
```diff
@@ -17,6 +17,9 @@ along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
 
 // https://docs.ens.domains/contract-developer-guide/writing-a-resolver
 // resolver interface Ids
```
Comment (Muhammad-Altabba) [inline packages/web3-eth-ens/src/config.ts]: Could you please paraphrase and edit this comment and use it inside your newly added comment?

File: packages/web3-eth-ens/src/config.ts [hunk 2]
```diff
@@ -27,6 +30,9 @@ export const interfaceIds: { [T: string]: string } = {
 };
 
 // functions list supported in resolver interfaces
```
Comment (Muhammad-Altabba) [inline packages/web3-eth-ens/src/config.ts]: (Same as above)
Could you please paraphrase and edit this comment and use it inside your newly added comment?

File: packages/web3-eth-ens/src/ens.ts [hunk 3]
```diff
@@ -39,6 +39,9 @@ import { registryAddresses } from './config';
 import { Resolver } from './resolver';
 
 export class ENS extends Web3Context<EthExecutionAPI & Web3NetAPI> {
```
Comment (Muhammad-Altabba) [inline packages/web3-eth-ens/src/ens.ts:45]: We need a description for the class, please.

Comment (Muhammad-Altabba) [reply packages/web3-eth-ens/src/ens.ts:45]: Additionally, comment for the constructor for its parameters and sample usage, please.

File: packages/web3-eth-ens/src/config.ts [hunk 4]
```diff
@@ -36,6 +42,9 @@ export const methodsInInterface: { [T: string]: string } = {
 	contenthash: 'contenthash',
 };
 
+/**
+ * An object holding the addressed of the ENS registries on the different networks (mainnet, ropsten, rinkeby, goerli).
+ */
 export const registryAddresses: { [T: string]: string } = {
 	main: '0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e',
 	ropsten: '0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e',
```
Comment (jdevcs) [inline packages/web3-eth-ens/src/config.ts]: Its not in scope of this issue but some of these networks (like ropsten, and rinkeby  ) are deprecated could you update this list?

**Review by jdevcs — APPROVED** [review]
lgtm, except changes required for existing comments.

**Review by avkos — APPROVED** [review]

**Pull request merged**